test/gtest/graphapi_mha_cpp_common.hpp

```diff
@@ -0,0 +1,444 @@
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "mha_helper.hpp"
+#include "gtest_common.hpp"
+#include "../verify.hpp"
+#include "../get_handle.hpp"
+#include "../tensor_holder.hpp"
+#include "../workspace.hpp"
+
+#include <gtest/gtest.h>
+
+#include <miopen/graphapi/execution_plan.hpp>
+#include <miopen/graphapi/matmul.hpp>
+#include <miopen/graphapi/opgraph.hpp>
+#include <miopen/graphapi/pointwise.hpp>
+#include <miopen/graphapi/reduction.hpp>
+#include <miopen/graphapi/rng.hpp>
+#include <miopen/graphapi/util.hpp>
+#include <miopen/graphapi/variant_pack.hpp>
+
+#include <numeric>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+#include <variant>
+
+namespace gr = miopen::graphapi;
+
+namespace mha_graph_test {
+class MhaGraphTestBase : public testing::TestWithParam<std::tuple<int, int, int, int, float>>
+{
+
+protected:
+    struct TensorData
+    {
+        using TensFlt = tensor<float>;
+        using TensI64 = tensor<int64_t>;
+
+        gr::Tensor* mTensPtr;
+        std::variant<TensFlt, TensI64> mCpuTensor;
+        miopen::Allocator::ManageDataPtr mGpuBuf;
+
+        explicit TensorData(gr::Tensor* tens_ptr) : mTensPtr(tens_ptr), mCpuTensor()
+        {
+            assert(mTensPtr);
+            const auto& d = mTensPtr->getDimensions();
+            std::vector<size_t> dims(d.begin(), d.end());
+            if(auto dt = mTensPtr->getDataType(); dt == miopenFloat)
+            {
+                mCpuTensor = TensFlt{dims};
+            }
+            else if(dt == miopenInt64)
+            {
+                mCpuTensor = TensI64{dims};
+            }
+            else
+            {
+                MIOPEN_LOG_FATAL("Unsupported data type for tensor" << dt);
```

Comment:
```suggestion
                FAIL() << "Unsupported data type for tensor" << dt;
```
Constructor does not care about return value.

Reply:
FYI: Well, the constructor does care, as in I get a build error due to return value. I didn't bother investigating it further and just replaced it  with `MIOPEN_FRIENDLY_FAIL`.